client/components/data/query-jetpack-credentials/index.jsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingJetpackCredentials } from 'state/selectors';
+import { fetchCredentials } from 'state/jetpack/credentials/actions';
+
+class QueryJetpackCredentials extends Component {
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.request( nextProps );
+		}
+	}
+
+	request( props ) {
+		if ( props.requestingCredentials ) {
```

Comment:
I think there's a logic issue here with the default value provided by `isRequestingJetpackCredentials`:
- initially, if there are no credentials requested, `isRequestingJetpackCredentials` will return `null`, which will make this request credentials.
- however, on `JETPACK_CREDENTIALS_REQUEST_FAILURE` or `JETPACK_CREDENTIALS_REQUEST_SUCCESS`, `isRequestingJetpackCredentials` will return `false`, which will make this go and request again.